### ISSUE TITLE
Implement `script.addPreloadScript` and `script.removePreloadScript`

### ISSUE THREAD

**Issue by sadym-chromium** [GoogleChromeLabs/chromium-bidi#293]
Implement suggestion in https://github.com/w3c/webdriver-bidi/pull/292, which can be directly mapped to [`Page.addScriptToEvaluateOnNewDocument`](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-addScriptToEvaluateOnNewDocument).

e2e tests are included in the task.

**Out of scope**: channels

## Tracking

- [x] `crypto` document: [go/webdriver:bidi-import-crypto-module](http://go/webdriver:bidi-import-crypto-module)
- [x] `crypto` implementation not depending on external module
- [x] mock/stub unit tests for preload script storage
- [x] add unit tests
- [x] fix last round of WPT tests
- [x] preload scripts should load in new contexts/tabs (`test_addPreloadScriptGlobally_loadedInNewContexts` E2E test should pass): https://github.com/GoogleChromeLabs/chromium-bidi/pull/673
- [x] make changes to CDP target w.r.t. unblocking order
- [x] OOPIF edge cases
- [x] ensure to account for parent context when `context` is provided in `add preload script`
- [x] ~~deduplicate CDPsession IDs when adding new preload scripts with a given `context` (they should be unique)~~
- [x] make WPT tests pass
- [x] make all E2E tests pass (`test_preload_script.py`)
- [x] ~~Add E2E test to expose race condition between [`CdpTarget.create` -> `void cdpTarget.#unblock()`](https://github.com/GoogleChromeLabs/chromium-bidi/blob/43882f46dca6e203b0bdbb12c4aa9d79a6a3895b/src/bidiMapper/domains/context/cdpTarget.ts#L55) and adding preload script: follow-up of https://github.com/GoogleChromeLabs/chromium-bidi/pull/673#discussion_r1184848960~~: superseded by item below 
  - [x] Upstream: [Issue 1449849: `Page.addScriptToEvaluateOnNewDocument` added before `Runtime.runIfWaitingForDebugger` should be run](https://bugs.chromium.org/p/chromium/issues/detail?id=1449849)
  - [x] Apply it in the codebase (`runImmediately`): https://github.com/GoogleChromeLabs/chromium-bidi/pull/919
- [x] support arguments / channels
- [x] support sandbox: https://github.com/GoogleChromeLabs/chromium-bidi/pull/978
- [x] reland https://github.com/GoogleChromeLabs/chromium-bidi/pull/879: https://github.com/GoogleChromeLabs/chromium-bidi/pull/951, https://github.com/GoogleChromeLabs/chromium-bidi/pull/983
- [x] add example script: https://github.com/GoogleChromeLabs/chromium-bidi/pull/985
- [ ] Extend `AddPreloadScript` with `ContextId` parameter https://github.com/GoogleChromeLabs/chromium-bidi/issues/1353

**Comment by thiagowfx** [comment]
- KR: Prototype “Bootstrap script” scenario
- Priority: P1
- Note: Scenario is focused on adding critical functionality for Puppeteer and other testing frameworks: the ability to run a script before any other scripts are run on a page. https://github.com/GoogleChromeLabs/chromium-bidi/milestone/15

**Comment by jrandolf** [comment]
We've reverted [Avoid deadlock in Target initialization (](https://github.com/GoogleChromeLabs/chromium-bidi/commit/d81f440040dbfd15896bb0f2a4988564cc9b1167)https://github.com/GoogleChromeLabs/chromium-bidi/pull/752[)](https://github.com/GoogleChromeLabs/chromium-bidi/commit/d81f440040dbfd15896bb0f2a4988564cc9b1167) because of E2E test breakage

**Comment by Lightning00Blade** [comment]
Closing the Umbrella bug the left issue will be track on its seprate bug.
